backends/tfhe-cuda-backend/cuda/include/device.h

```diff
@@ -13,13 +13,18 @@ extern "C" {
 
 #define check_cuda_error(ans)                                                  \
   { cuda_error((ans), __FILE__, __LINE__); }
+
+#ifdef DEBUG
 inline void cuda_error(cudaError_t code, const char *file, int line) {
   if (code != cudaSuccess) {
     std::fprintf(stderr, "Cuda error: %s %s %d\n", cudaGetErrorString(code),
                  file, line);
     std::abort();
   }
 }
+#else
+inline void cuda_error(cudaError_t code, const char *file, int line) {}
+#endif
```

Comment:
So in release mode we would just let the program continue even when an error occured ? isn't that a bit dangerous ?

Reply:
Yes,in release mode we wouldn't be  explicitly checking for exceptions during kernel launches (as invalid configurations) or memory operations (as copies to device).
The motivation is to avoid possible overheads.

Not sure about dangers. Would that be bad? Maybe we should rethink @agnesLeroy

Reply:
Was the overhead measured ?

Are we sure the tests we run do check that all possible error case are properly checked and that when running in release mode, we would not skip an error and continue in an invalid state ?

If I'm correct the CI tests run in release mode, so the CI would not trigger errors

Reply:
Ok, this change does not impact performance so I will remove it, since it may also impact correctness / security. 

The debug/release modes during compiling are useful so I will keep the PR only with that.